src/main/java/org/example/controller/FacultyController.java

```diff
@@ -0,0 +1,66 @@
+package org.example.controller;
+
+import org.example.model.Faculty;
+import org.example.service.FacultyService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.Collections;
+
+@RestController
+@RequestMapping("faculty")
+public class FacultyController {
+
+    private final FacultyService facultyService;
+
+    public FacultyController(FacultyService facultyService) {
+        this.facultyService = facultyService;
+    }
+
+
+    @GetMapping("{id}")
+    public ResponseEntity<Faculty> getFaculty(@PathVariable Long id) {
+        Faculty faculty = facultyService.findFaculty(id);
+        if (faculty == null) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(faculty);
+    }
+
+    @GetMapping("/all")
+    public ResponseEntity<Collection<Faculty>> getAllFaculties() {
+        return ResponseEntity.ok(facultyService.getAllFaculties());
+    }
+
+    @GetMapping("/color")
+    public ResponseEntity<Collection<Faculty>> getFacultiesByColor(@RequestParam(required = false) String color) {
+        if (color != null && !color.isBlank()) {
+            return ResponseEntity.ok(facultyService.getFilteredFacultiesByColor(color));
+        }
+        return ResponseEntity.ok(Collections.emptyList());
+    }
+
+    @PostMapping()
+    public Faculty createFaculty(@RequestBody Faculty faculty) {
+        return facultyService.createFaculty(faculty);
+    }
+
+    @PutMapping
+    public ResponseEntity<Faculty> editFaculty(@RequestBody Faculty faculty) {
+        Faculty foundeFaculty = facultyService.editFaculty(faculty);
+        if (foundeFaculty == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+        return ResponseEntity.ok(foundeFaculty);
+    }
+
+    @DeleteMapping("{id}")
+    public ResponseEntity deleteFaculty(@PathVariable Long id) {
```

Comment:
```suggestion
    public ResponseEntity<?> deleteFaculty(@PathVariable Long id) {
```
дженерик надо писать